### PR TITLE
[BigNLP] Add CI Tests for Megatron GPT

### PR DESCRIPTION
**Pull request by ericharper**

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -64,7 +63,7 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             seed=self.cfg.get('seed', 1234),
         )
 
-        if not self.cfg.get('fused_bf16'):
+        if self.cfg.get('precision') != 'bf16':
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:66]: I don't think we have access to the train config here right? Isn't this the model config that doesn't have the precision param?

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 2]
```diff
@@ -100,8 +99,7 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             fp16_lm_cross_entropy=cfg.get('fp16_lm_cross_entropy', False),
             use_cpu_initialization=cfg.get('use_cpu_initialization', False),
             hidden_dropout=cfg.get('hidden_dropout', 0.1),
-            fused_fp16=cfg.get('fused_fp16', False),
-            fused_bf16=cfg.get('fused_bf16', False),
+            precision=cfg.get('precision', 16),
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:102]: Same comment here - maybe we want to add precision to the model config via ${trainer.precision}?

**Review by MaximumEntropy — COMMENTED** [review]
LGTM! Minor comments about adding precision to the model config.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:102]: Yup, that's right. I'll make the change.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 1 above)
Comment (ericharper) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:66]: We add it to the config in the training script.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:102]: I ended up not being able to use interpolation as the ${trainer.precision} could not be found when PTL saves the hparams. I added a comment explaining why we add the precision to the model config in the training script.

**Review by okuchaiev — APPROVED** [review]

**Pull request merged**